mlir/lib/Dialect/LLVMIR/Transforms/TypeConsistency.cpp

```diff
@@ -524,9 +525,9 @@ static void splitIntegerStore(const DataLayout &dataLayout, Location loc,
 
     // We create an `i8` indexed GEP here as that is the easiest (offset is
     // already known). Other patterns turn this into a type-consistent GEP.
-    auto gepOp =
-        rewriter.create<GEPOp>(loc, address.getType(), rewriter.getI8Type(),
-                               address, ArrayRef<GEPArg>{currentOffset});
+    auto gepOp = rewriter.create<GEPOp>(
+        loc, address.getType(), rewriter.getI8Type(), address,
+        ArrayRef<GEPArg>{static_cast<int32_t>(currentOffset)});
```

Comment:
Nit: It might be easier to just change the type of `currentOffset`, or even the `storeOffset` parameter.

Reply:
it participates in other expressions involving `unsigned` values (e.g. as "accumulator" below) so figured the cast here is cleaner.